client/consensus/grandpa/authorities.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package grandpa
+
+import (
+	"errors"
+	"fmt"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+)
+
+// AuthorityList A list of Grandpa authorities with associated weights.
+// TODO migrate this type and associated functions to this package
+type AuthorityList []types.Authority
+
+// PendingChange A pending change to the authority set.
+//
+// This will be applied when the announcing block is at some depth within
+// the finalized or unfinalized chain.
+type PendingChange struct {
+	nextAuthorities AuthorityList
+	delay           uint
+	canonHeight     uint
+	canonHash       common.Hash
+	delayKind       DelayKind
+}
```

Comment:
can we not use generics here?  Substrate defines a `H` type and `N` type to represent hash and number.  
```
pub struct PendingChange<H, N> {
	/// The new authorities and weights to apply.
	pub(crate) next_authorities: AuthorityList,
	/// How deep in the chain the announcing block must be
	/// before the change is applied.
	pub(crate) delay: N,
	/// The announcing block's height.
	pub(crate) canon_height: N,
	/// The announcing block's hash.
	pub(crate) canon_hash: H,
	/// The delay kind.
	pub(crate) delay_kind: DelayKind<N>,
}
```
You can use `constraints.Unsigned` for `H` and `comparable` for `N`.